libarchive/archive_entry.h

```diff
@@ -262,6 +262,7 @@ __LA_DECL const wchar_t	*archive_entry_gname_w(struct archive_entry *);
 __LA_DECL const char	*archive_entry_hardlink(struct archive_entry *);
 __LA_DECL const char	*archive_entry_hardlink_utf8(struct archive_entry *);
 __LA_DECL const wchar_t	*archive_entry_hardlink_w(struct archive_entry *);
+__LA_DECL int		 archive_entry_hardlink_is_set(struct archive_entry *);
```

Comment:
This was previously suggested in the linked to PR. I just noticed that there's an earlier comment in this file that says:

> Unset string fields return NULL, non-string fields have _is_set() functions to test whether they've been set

This kind of breaks that, however I think it's still valuable since it's possible for some subset of string functions to return null even if others return non-null